api/src/org/apache/cloudstack/api/command/user/vm/UpdateVMCmd.java

```diff
@@ -136,13 +136,13 @@ public String getInstanceName() {
         return instanceName;
     }
 
-    public Map getDetails() {
+    public Map<String, String> getDetails() {
```

Comment:
The CommandType.MAP API args are usually map of object returning objects; in practice AFAIR all map args are string->string. Either way, I don't understand this change in getDetails(), this at least will result in returning a new Map, so likely the author wants their details to be sort of read-only? If so, why not use final?